channel.go

```diff
@@ -200,23 +261,59 @@ func (c *channel) recvMsgs() {
 	}
 }
 
-func (c *channel) reconnect() {
-	c.streamMut.Lock()
-	defer c.streamMut.Unlock()
+func (c *channel) tryReconnect() {
```

Comment:
I wonder if some of the method names should be named something with `stream`. I wonder if it would be a clearer logical separation if we kept the logic of dialing to get a connection object as methods on the `RawNode` struct, and the stream-related "objects" in the `channel` struct.

Reply:
channel.tryConnect() was initially created as a safety measure, but I agree that it is now more confusing than helping. The logic in channel.connect() could thus be moved to newChannel() instead. Then we could rename channel.tryConnect() to channel.connect(). This is not problematic as both newChannel() and channel.connect() is only used node.connect().

I don't think it is trivial to move the atomic flag to node.go because it needs to be checked every time a new message is sent. Either it would require a large refactor or that the channel is calling several methods on the node. In my opinion it would make more sense to move the dial method to the channel.

I have tried to rename some of the methods in the last commit to make it more clear/less confusing.